src/taskgraph/optimize/strategies.py

```diff
@@ -40,9 +40,10 @@ def should_replace_task(self, task, params, deadline, arg):
                     continue
 
                 return task_id
-            except KeyError:
-                # go on to the next index path
-                pass
+            except KeyError as e:
+                logger.warning(
```

Comment:
This should definitely be a `debug` level, if not `trace` :)

Reply:
Fixed